sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/client.py

```diff
@@ -895,6 +934,58 @@ def receive_message_batch(self, **kwargs):
         """
         return self._do_retryable_operation(self._receive_message_batch_impl, **kwargs)
 
+    def receive_messages_iter(self, on_message_received=None):
+        """Receive messages by generator. Messages returned in the generator have already been
+        accepted - if you wish to add logic to accept or reject messages based on custom
+        criteria, pass in a callback.
+
+        :param on_message_received: A callback to process messages as they arrive from the
+         service. It takes a single argument, a ~uamqp.message.Message object.
+        :type on_message_received: callable[~uamqp.message.Message]
+        """
+        self._message_received_callback = on_message_received
+        return self._message_generator()
+
+    def _message_generator(self):
+        """Iterate over processed messages in the receive queue.
+
+        :rtype: generator[~uamqp.message.Message]
+        """
+        auto_complete = self.auto_complete
+        self.auto_complete = False
+        self._timeout_reached = False
+        receiving = True
+        message = None
+        self._last_activity_stamp = time.time()
+        try:
+            while receiving and not self._timeout_reached:
+                if not self._running_iter:
+                    self._last_activity_stamp = time.time()
+                self._running_iter = True
+                if self._timeout > 0:
+                    if time.time() - self._last_activity_stamp >= self._timeout:
+                        self._timeout_reached = True
+
+                if not self._timeout_reached:
+                    receiving = self.do_work()
+
+                while not self._received_messages.empty():
+                    message = self._received_messages.get()
+                    self._last_activity_stamp = time.time()
+                    self._received_messages.task_done()
+                    yield message
+                    self._complete_message(message, auto_complete)
+
+        finally:
+            if self._shutdown:
+                self.close()
+
+    def _complete_message(self, message, auto): # pylint:disable=no-self-use
+        if not message or not auto:
+            return
```

Comment:
We don't need this do we?
I don't think ServiceBus ever needs us to complete a message on behalf of users, so I would just remove it entirely.